lib/coinbase/webhook.rb

```diff
@@ -0,0 +1,167 @@
+          Coinbase::Webhook.new(webhook)
+        end
+      end
+
+      private
+
+      def webhooks_api
+        Coinbase::Client::WebhooksApi.new(Coinbase.configuration.api_client)
+      end
+
+      def fetch_webhooks_page(page)
+        webhooks_api.list_webhooks({ limit: DEFAULT_PAGE_LIMIT, page: page })
+      end
+    end
+
+    # Initializes a new Webhook object.
+    #
+    # @param model [Coinbase::Client::Webhook] The underlying Webhook object.
+    # @raise [ArgumentError] If the model is not a Coinbase::Client::Webhook.
+    def initialize(model)
+      raise ArgumentError, 'model must be a Webhook' unless model.is_a?(Coinbase::Client::Webhook)
+
+      @model = model
+    end
+
+    # Returns the ID of the webhook.
+    #
+    # @return [String] The ID of the webhook.
+    def id
+      @model.id
+    end
+
+    # Returns the network ID associated with the webhook.
+    #
+    # @return [String] The network ID of the webhook.
+    def network_id
+      @model.network_id
+    end
+
+    # Returns the notification URI of the webhook.
+    #
+    # @return [String] The URI where notifications are sent.
+    def notification_uri
+      @model.notification_uri
+    end
+
+    # Returns the event type of the webhook.
+    #
+    # @return [String] The type of event the webhook listens for.
+    def event_type
+      @model.event_type
+    end
+
+    # Returns the event filters applied to the webhook.
+    #
+    # @return [Array<Hash>] An array of event filters used by the webhook.
+    def event_filters
+      @model.event_filters
+    end
+
+    # Updates the webhook with a new notification URI.
+    #
+    # @param notification_uri [String] The new URI for webhook notifications.
+    # @return [self] Returns the updated Webhook object.
+    #
+    # @example Update the notification URI of a webhook
+    #   webhook.update(notification_uri: 'https://new-url.com/callback')
+    def update(notification_uri:)
+      serialized_event_filters = event_filters.map do |filter|
+        {
+          contract_address: filter['contract_address'],
+          from_address: filter['from_address'],
+          to_address: filter['to_address']
+        }.compact
+      end
+
+      model = Coinbase.call_api do
+        webhooks_api.update_webhook(
+          id,
+          update_webhook_request: {
+            network_id: network_id,
+            notification_uri: notification_uri,
+            event_type: event_type,
+            event_filters: serialized_event_filters
+          }
+        )
+      end
+
+      @model = model
+
+      self
+    end
+
+    # Deletes the webhook.
+    #
+    # @return [self] Returns the Webhook object with nil attributes.
+    #
+    # @example Delete a webhook
+    #   webhook.delete
+    def delete
+      Coinbase.call_api do
+        webhooks_api.delete_webhook(
+          id
+        )
+      end
+
+      @model = nil
+
+      self
+    end
+
```

Comment:
```suggestion

    # Returns a String representation of the Webhook.
    # @return [String] a String representation of the Webhook
    def to_s
      Coinbase.pretty_print_object(
        self.class,
        id: @model.id,
        network_id: @model.network_id,
        event_type: @model.event_type,
        notification_uri: @model.notification_uri,
        event_filters: @model.event_filters.map(&:to_hash).to_json
      )
    end

    # Same as to_s.
    # @return [String] a String representation of the Webhook
    def inspect
      to_s
    end
```

Can you squash your commits, rebase on top of `v0.0.16` and then add these helpers to properly render this object in console? Thanks!